### DISCUSSION TITLE
Add an Action to Insert a Link

### DISCUSSION THREAD

**Discussion by caracolmzd** [GitJournal/GitJournal#755] [Feature Requests]
When adding a file-link, on my android keyboard, it requires changing character mode 3 times, if I plan ahead, and more times if I type naturally....
1. Select Symbols, and then select the alternate symbols to get square-brackets
1. Switch back to primary symbols for Parens
1. Switch back to text to type the link name and URL

I could see having a link icon next to the insert attachment icon in the file editor. It could simply insert the brackets for a link, either with or without place-holders.

If text is selected when insert-link is clicked, text is used for the link text.

A bonus feature would be similar to what Typora provides. I realize permissions to read the clipboard might be an obstacle to this...
1. When inserting a link template
1. Check the contents of the clipboard for a pattern matching a URL or file-path
1. If not matched, insert-link action is un-changed
1. If matched, use clipboard contents for the link URL.

**Comment by bergie** [comment]
This would be a great place to allow searching for notes inside GitJournal to link to. Now linking to another note requires quite a few steps (open the other note, hamburger menu, edit file name, select all, copy) even before the keyboard gymnastics described above.

Related to #210 and #177 but for regular Markdown links instead of wiki links.

**Comment by nottrobin** [comment]
I'd love this! I just paid for a year GitJournal, and it's nearly perfect. Really a shortcut to add links is about the only thing missing for me. Please implement it if you can! :)
